### DISCUSSION TITLE
Seems Issue with grouping specs defined in suites.

### DISCUSSION THREAD

**Discussion by anupamtomar** [webdriverio/webdriverio#11471] [Q&A]
Let's say I have designed the below suite, and I have defined maxInstances: 3

```
"suites": {
        someSuite1: [
            [
                "./test/specs/test_register.js",
                "./test/specs/test_login.js",
            ],
            [
                "./test/specs/test_forgotPassword.js",
                "./test/specs/test_changePassword.js",
            ],
            [
                "./test/specs/test_updateProfile.js",
                "./test/specs/test_deleteProfile.js",
            ]
        ],
},
```
Now as per my understanding from reading the below document, it should start the 3 separate and concurrent workers since we have grouped Test Specs in 3 groups, but the specs inside these groups will run sequentially in their respective instances, but actually what happening is that the other spec file start executing on the same instance after some time before even closing the firstly initiated spec file of same group due to which it is failing and overwriting the commonly used reference variables.
Moreover, the problem becomes more when using mocha hooks inside the describe block of the spec files.

On the other hand, the same test specs are working fine if executing standard sequentially with 1 worker defined or executing with standard parallel execution with multiple workers defined. ('Standard' I mean when not grouping specs defined in suites).

Reference doc link-
https://webdriver.io/docs/organizingsuites/#:~:text=Grouping%20Test%20Specs%20To%20Run,run%20in%20a%20single%20instance.

**Comment by christian-bromann** [comment]
@anupamtomar interesting observation. Can you maybe create a minimal reproducible example?
